collection_script_draft1.py

```diff
@@ -0,0 +1,409 @@
+            # print this to the console indicating the video is skipped
+            print(
+                video_duration_in_seconds, " Seconds. Video Skipped for being too Long!"
+            )
+            # onto the next url
+            continue
+
+        # creates a new dir by the index of the url (marks what video we are playing) in the list_of_urls
+        Path(new_dir).mkdir(parents=False, exist_ok=True)
+
+        # stores the current state (integer) of the YouTube player as per YouTube's pre-set API
+        # used to check if the video is playing, paused, buffering etc.
+        video_playing: int = driver.execute_script(
+            "return document.getElementById('movie_player').getPlayerState()"
+        )
+        # sleep for 0.5 seconds for the script to execute
+        time.sleep(0.5)
+
+        # checks the state of the player -- if the video has not started to play yet
+        if video_playing != 1:
+            # prints a confirmation message to the console
+            print("Video has now started playing")
+            # register an onSpace event by pressing the space button to play the video
+            movie_player.send_keys(Keys.SPACE)
+        else:
+            # in the event it is already playing simply print this
+            print("Video is already playing")
+
+        # initiating an infinite loop
+        while True:
+            # time.sleep(0.5)
+            # getting the state of the video
+            video_playing: int = driver.execute_script(
+                "return document.getElementById('movie_player').getPlayerState()"
+            )
+
+            # getting the present status of the ad (if any)
+            ad_playing: int = driver.execute_script(
+                "return document.getElementsByClassName('ad-showing').length"
+            )
+            # getting the current duration of the main video played so far
+            video_played_in_seconds: int = driver.execute_script(
+                'return document.getElementById("movie_player").getCurrentTime()'
+            )
+
+            # UH-OH SOMETHING IS FISHY
+            # if the video is not playing and the ad is not playing
+            if video_playing != 1 and not ad_playing:
+                # if the video playing has not ended
+                if video_playing != 0:
+                    # press space to play the video
+                    movie_player.send_keys(Keys.SPACE)
+
+            # if the ad is playing
+            if ad_playing:
+                # bool variable to keep track of whether the ad is playing or not
+                ad_just_played = True
+
+                # checks if the ad is playing or not -- if it is paused or not
+                is_add_playing = driver.execute_script(
+                    'return document.getElementsByClassName("ytp-play-button ytp-button")[0].title'
+                )
+                # if the ad is playing
+                if is_add_playing != "Pause (k)":
```

Comment:
This was capturing the small play button on the bottom left of the player. Doesnt exist on mobile